scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -220,37 +255,92 @@ class ScalametaParser(input: Input, dialect: Dialect) { parser =>
           next.is[CloseDelim] &&
           next.pos.startLine > curr.pos.endLine
       if (curr.isNot[Trivia] && !isTrailingComma) {
-        parserTokens += curr
-        parserTokenPositions += currPos
-        val sepRegions1 = {
-          if (curr.is[LeftParen]) ')' :: sepRegions
-          else if (curr.is[LeftBracket]) ']' :: sepRegions
+
+        var sepRegions = if (indentedRegion(sepRegionsP) && currentIndent >= 0) {
+          var sepRegionsI = sepRegionsP
+          while (indentedRegion(sepRegionsI) && sepRegionsI.head.indent > currentIndent) {
+            parserTokens += new Indentation.Outdent(curr.input, curr.dialect, curr.start, curr.end)
+            parserTokenPositions += currPos
+            parserTokens += new LF(curr.input, curr.dialect, curr.start)
+            parserTokenPositions += currPos
+            sepRegionsI = sepRegionsI.tail
+          }
+          sepRegionsI
+        } else sepRegionsP
+
+        sepRegions = if (addRegion > 0) {
+          if (!sepRegions.isEmpty && sepRegions.head.closing == '{')
+            SepRegion('E', addRegion) :: sepRegions.tail
```

Comment:
Why is the differentiation needed?

Reply:
because sometimes you want to check: "Am I inside enum?"
and sometimes: "Am I inside indented region?"
You could change it completely to sort of flags and you could in case of 'indented enum' add INDENT+ENUM but I don't know if this is better approach